src/components/CameraError/index.tsx

```diff
@@ -18,72 +19,60 @@ type Props = {
   renderFallback: RenderFallbackProp
 } & WithTrackingProps
 
-type State = {
-  isDimissed: boolean
-}
-
-export default class CameraError extends Component<Props, State> {
-  state = {
-    isDimissed: false,
-  }
-
-  componentDidUpdate(prevProps: Props): void {
-    if (prevProps.error.name !== this.props.error.name) {
-      this.setState({ isDimissed: false })
-    }
-  }
+const CameraError: FunctionComponent<Props> = ({
+  error,
+  hasBackdrop,
+  isDismissible,
+  renderFallback,
+  trackScreen,
+}: Props) => {
+  const [isDismissed, setIsDismissed] = useState(false)
 
-  trackFallbackClick = (): void => {
-    const { type, name } = this.props.error
+  const trackFallbackClick = useCallback(() => {
+    const { type, name } = error
     if (type === 'warning') {
       const uiAlertName = ErrorNameToUIAlertMapping[name]
       const properties: AnalyticsEventProperties = uiAlertName
         ? { ui_alerts: { [uiAlertName]: type } }
         : {}
-      this.props.trackScreen('fallback_triggered', properties)
+      trackScreen('fallback_triggered', properties)
     }
-  }
-
-  handleDismiss = (): void => this.setState({ isDimissed: true })
-
-  render(): h.JSX.Element | null {
-    const {
-      error,
-      hasBackdrop,
-      renderFallback,
-      isDismissible,
-      trackScreen,
-    } = this.props
+  }, [error, trackScreen])
 
-    if (this.state.isDimissed) {
-      return null
-    }
+  useEffect(() => {
+    setIsDismissed(false)
```

Comment:
shouldn't we test the error name equality ? (Unsure if the case happens that a new error object pops up with the same name)

Reply:
You're right. I don't know if it happens, but if the component is re-rendered and the reference to the `error` object changes, it will dismiss the error message.

I can just replace `error` with `error.name` in the dependency array to get the exact same behaviour as before.